src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerFactoryExtensions.cs

```diff
@@ -43,12 +42,19 @@ public static class ApplicationInsightsLoggerFactoryExtensions
         /// </summary>
         /// <param name="factory"></param>
         /// <param name="filter"></param>
-        /// <param name="client">The instance of <see cref="TelemetryClient"/> to use for logging.</param>
+        /// <param name="serviceProvider">The instance of <see cref="IServiceProvider"/> to use for logging.</param>
         public static ILoggerFactory AddApplicationInsights(
             this ILoggerFactory factory,
-            TelemetryClient client,
+            IServiceProvider serviceProvider,
             Func<string, LogLevel, bool> filter)
         {
+            var client = serviceProvider.GetService<TelemetryClient>();
+            var debugLoggerControl = serviceProvider.GetService<DebugLoggerControl>();
+            if (debugLoggerControl != null)
+            {
+                debugLoggerControl.EnableDebugLogger = false;
```

Comment:
Just curious so I understand this, when DebugLoggerControl is created the flag is defaulted to true but then here when adding App Insights it is set to false, can you explain the flow for enabling and disabling the debug logger?